manifests/params.pp

```diff
@@ -19,8 +19,17 @@
     $datadir    = '/etc/puppetlabs/puppet/hieradata'
     $owner      = 'pe-puppet'
     $group      = 'pe-puppet'
-    $provider   = 'pe_gem'
     $confdir    = '/etc/puppetlabs/puppet'
+    if $::puppetversion =~ /3.7/ {
+      $provider = 'pe_puppetserver_gem'
+      # 3.7 also needs /opt/puppet/bin/eyaml
+      exec { 'install pe_gem':
```

Comment:
This should be in `eyaml.pp`.  Furthermore, rather than having an exec, it'd be a better idea to have another package resource:
```puppet
if $::is_pe {
  package { 'hiera-eyaml command line':
    name     => 'hiera-eyaml',
    ensure   => installed,
    provider => 'pe_gem',
  }
}
```